### PR TITLE
Refactor variant and genotype annotations

### PR DESCRIPTION
**Pull request by heuermh**
Update per https://github.com/bigdatagenomics/bdg-formats/pull/90.  Currently compiles but fails 14 unit tests.

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotationConverter.scala [hunk 1]
```diff
@@ -21,11 +21,7 @@ import org.apache.avro.Schema
 import org.apache.avro.specific.SpecificRecord
 import htsjdk.variant.variantcontext.VariantContext
 import htsjdk.variant.vcf._
-import org.bdgenomics.formats.avro.{
-  DatabaseVariantAnnotation,
-  Genotype,
-  VariantCallingAnnotations
-}
+import org.bdgenomics.formats.avro.{ Genotype, GenotypeAnnotation, VariantAnnotation }
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotationConverter.scala]: How long is this line? I generally break the imports into multiple lines if I'm importing more than 4 things, or if the import line is >80 characters.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotationConverter.scala [hunk 2]
```diff
@@ -191,31 +149,22 @@ private[converters] object VariantAnnotationConverter extends Serializable {
...
-   * VariantCallingAnnotations schema.
-   */
-  lazy val VCF2VariantCallingAnnotations: Map[String, (Int, Object => Object)] =
-    createFieldMap(INFO_KEYS, VariantCallingAnnotations.getClassSchema)
-
-  /**
    * Mapping between VCF format field names and field IDs in the Genotype schema.
    */
-  lazy val VCF2GenotypeAnnotations: Map[String, (Int, Object => Object)] =
+  lazy val VCF2Genotype: Map[String, (Int, Object => Object)] =
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotationConverter.scala]: lower caps on `vcf` and s/2/To/g?

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotationConverter.scala [hunk 3]
```diff
@@ -191,31 +149,22 @@ private[converters] object VariantAnnotationConverter extends Serializable {
...
+    createFieldMap(FORMAT_KEYS, GenotypeAnnotation.getClassSchema)
 
   /**
-   * Mapping between VCF info field names and DatabaseVariantAnnotation schema
-   * field IDs for all database specific fields.
+   * Mapping betweem VCF info field names and fields IDs in the VariantAnnotation schema.
    */
-  lazy val VCF2DatabaseAnnotations: Map[String, (Int, Object => Object)] = createFieldMap(EXTERNAL_DATABASE_KEYS,
-    DatabaseVariantAnnotation.getClassSchema)
+  lazy val VCF2VariantAnnotation: Map[String, (Int, Object => Object)] =
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotationConverter.scala]: Ditto here RE: caps, 2.

**Comment by jpdna** [comment]
When trying to build the code at huerermh:annotations  abe149fd104 I get an error:
````
[ERROR] /home/jp/Berk/work/issues/PR1078/adam/adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala:498: error: value setVariantIsPassing is not a member of org.bdgenomics.formats.avro.GenotypeAnnotation.Builder
````
because there is no `variantIsPassing` field in `GenotypeAnnotation` in  bdg-formats master at 5e81a9c.
Is there a different version of bdg-formats I should build/install first?

**Comment by fnothaft** [comment]
I think it needs to be updated for https://github.com/bigdatagenomics/bdg-formats/pull/88.

**Comment by heuermh** [comment]
Right, #1075 needs to be applied first

**Comment by jpdna** [comment]
@heuermh  do you have a VCF file with the ANN tag that you have used for testing?
I made an annotated file using SNPEFF over the ClinVar dataset, and then tried running:
```
val x1 = sc.loadVcfAnnotations("file:///home/jp/Berk/work/issues/PR1078/run1/snpeff.vcf")
val test = x1.take(1)
```
and I'm getting a NullPointerException that I am investigating.  
It would be useful to me to have a positive control to check that the above sc.loadVcfAnnotations works for well formed input VCF with ANN tag

**Comment by heuermh** [comment]
As mentioned above, this branch isn't ready to be reviewed:  Currently compiles but fails 14 unit tests.

**Comment by heuermh** [comment]
For the record, I'm not too good with having lots of different things open and unresolved at the same time, and I currently have four open pull requests in the same code path.  If I could get #1060 and #1069 merged soon, that would be helpful.

Then #1075 and this pull request need a 0.10.0 release of bdg-formats, which is waiting on doc fixes, see https://github.com/bigdatagenomics/bdg-formats/issues/92.

**Comment by jpdna** [comment]
> As mentioned above, this branch isn't ready to be reviewed:

Ah, no problem, ping when ready - I'll plan to stress test with annotations from snpEff/VEP

**Comment by heuermh** [comment]
Rebased and cherry-picked commit from #1075.  Now there are 15 unit test failures.  :)

**Comment by fnothaft** [comment]
We're closing this in favor of #1135, right @heuermh? Please reopen if I've incorrectly closed.

**Comment by heuermh** [comment]
Yes, thank you.

**Pull request closed without merging**